diff --git a/lib/command.js b/lib/command.js
index b35db1c..6defc94 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -25,7 +25,7 @@ class Command extends EventEmitter {
     this.options = [];
     this.parent = null;
     this._allowUnknownOption = false;
-    this._allowExcessArguments = true;
+    this._allowExcessArguments = false;
     /** @type {Argument[]} */
     this.registeredArguments = [];
     this._args = this.registeredArguments; // deprecated old name
diff --git a/tests/command.allowExcessArguments.test.js b/tests/command.allowExcessArguments.test.js
index 4659d0c..8967342 100644
--- a/tests/command.allowExcessArguments.test.js
+++ b/tests/command.allowExcessArguments.test.js
@@ -14,13 +14,13 @@ describe.each([true, false])(
       }
     }
 
-    test('when specify excess program argument then no error by default', () => {
+    test('when specify excess program argument then error by default', () => {
       const program = new commander.Command();
       configureCommand(program);
 
       expect(() => {
         program.parse(['excess'], { from: 'user' });
-      }).not.toThrow();
+      }).toThrow();
     });
 
     test('when specify excess program argument and allowExcessArguments(false) then error', () => {
@@ -53,14 +53,14 @@ describe.each([true, false])(
       }).not.toThrow();
     });
 
-    test('when specify excess command argument then no error (by default)', () => {
+    test('when specify excess command argument then error (by default)', () => {
       const program = new commander.Command();
       const sub = program.command('sub');
       configureCommand(sub);
 
       expect(() => {
         program.parse(['sub', 'excess'], { from: 'user' });
-      }).not.toThrow();
+      }).toThrow();
     });
 
     test('when specify excess command argument and allowExcessArguments(false) then error', () => {
