diff --git a/lib/winston/container.js b/lib/winston/container.js
index 1c6f114..891b6fb 100644
--- a/lib/winston/container.js
+++ b/lib/winston/container.js
@@ -40,7 +40,15 @@ module.exports = class Container {
 
       // Remark: Make sure if we have an array of transports we slice it to
       // make copies of those references.
-      options.transports = existing ? existing.slice() : [];
+      let trans;
+      if (Array.isArray(existing)) {
+        trans = existing.slice();
+      } else if (existing) {
+        trans = [existing];
+      } else {
+        trans = [];
+      }
+      options.transports = trans;
 
       const logger = createLogger(options);
       logger.on('close', () => this._delete(id));
