diff --git a/__tests__/regression.test.ts b/__tests__/regression.test.ts
index 6a8225c..85df103 100644
--- a/__tests__/regression.test.ts
+++ b/__tests__/regression.test.ts
@@ -34,25 +34,25 @@ describe('Regression Tests', () => {
     const chess = new Chess()
     const pgn = [
       '[SetUp "1"]',
-      '[FEN "7k/5K2/4R3/8/8/8/8/8 w KQkq - 0 1"]',
+      '[FEN "7k/5K2/4R3/8/8/8/8/8 w - - 0 1"]',
       '',
       '1. Rh6#',
     ]
     chess.loadPgn(pgn.join('\n'))
-    expect(chess.fen()).toBe('7k/5K2/7R/8/8/8/8/8 b KQkq - 1 1')
+    expect(chess.fen()).toBe('7k/5K2/7R/8/8/8/8/8 b - - 1 1')
   })
 
   it('Github Issue #85 (black) - SetUp and FEN should be accepted in loadPgn', () => {
     const chess = new Chess()
     const pgn = [
       '[SetUp "1"]',
-      '[FEN "r4r1k/1p4b1/3p3p/5qp1/1RP5/6P1/3NP3/2Q2RKB b KQkq - 0 1"]',
+      '[FEN "r4r1k/1p4b1/3p3p/5qp1/1RP5/6P1/3NP3/2Q2RKB b - - 0 1"]',
       '',
       '1. ... Qc5+',
     ]
     chess.loadPgn(pgn.join('\n'))
     expect(chess.fen()).toBe(
-      'r4r1k/1p4b1/3p3p/2q3p1/1RP5/6P1/3NP3/2Q2RKB w KQkq - 1 2'
+      'r4r1k/1p4b1/3p3p/2q3p1/1RP5/6P1/3NP3/2Q2RKB w - - 1 2'
     )
   })
 
diff --git a/src/chess.ts b/src/chess.ts
index b01c6bd..421da18 100644
--- a/src/chess.ts
+++ b/src/chess.ts
@@ -616,6 +616,8 @@ export class Chess {
       this._castling.b |= BITS.QSIDE_CASTLE
     }
 
+    this._updateCastlingRights()
+
     this._epSquare = tokens[3] === '-' ? EMPTY : Ox88[tokens[3] as Square]
     this._halfMoves = parseInt(tokens[4], 10)
     this._moveNumber = parseInt(tokens[5], 10)
