diff --git a/lib/request.js b/lib/request.js
index b756a7a..59d38e6 100644
--- a/lib/request.js
+++ b/lib/request.js
@@ -10,7 +10,7 @@ const accepts = require('accepts')
 const contentType = require('content-type')
 const stringify = require('url').format
 const parse = require('parseurl')
-const qs = require('querystring')
+
 const typeis = require('type-is')
 const fresh = require('fresh')
 const only = require('./only.js')
@@ -171,7 +171,7 @@ module.exports = {
   get query () {
     const str = this.querystring
     const c = this._querycache = this._querycache || {}
-    return c[str] || (c[str] = qs.parse(str))
+    return c[str] || (c[str] = Object.fromEntries(new URLSearchParams(str)))
   },
 
   /**
@@ -182,7 +182,11 @@ module.exports = {
    */
 
   set query (obj) {
-    this.querystring = qs.stringify(obj)
+    const params = new URLSearchParams()
+    for (const key of Object.keys(obj)) {
+      params.set(key, obj[key])
+    }
+    this.querystring = params.toString()
   },
 
   /**
