diff --git a/lib/command.js b/lib/command.js
index 4c5089a..6517747 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -48,6 +48,7 @@ class Command extends EventEmitter {
     this._aliases = [];
     this._combineFlagAndOptionalValue = true;
     this._description = '';
+    this._summary = undefined; // use summary for subcommand listing
     this._argsDescription = undefined; // legacy
     this._enablePositionalOptions = false;
     this._passThroughOptions = false;
