diff --git a/lib/winston.js b/lib/winston.js
index 2357d99..a14d831 100644
--- a/lib/winston.js
+++ b/lib/winston.js
@@ -50,6 +50,11 @@ exports.ExceptionHandler = require('./winston/exception-handler');
  * @type {Object}
  */
 exports.RejectionHandler = require('./winston/rejection-handler');
+/**
+ * Expose Logger class - allows extending it with custom functionality.
+ * @type {Function}
+ */
+exports.Logger = require('./winston/logger');
 /**
  * Expose core Logging-related prototypes.
  * @type {Container}
