diff --git a/src/List.test.ts b/src/List.test.ts
index 2f99193..5cf7019 100644
--- a/src/List.test.ts
+++ b/src/List.test.ts
@@ -66,4 +66,15 @@ describe('List', () => {
     // immutability check
     expect(List.sort(compare, arr)).not.toBe(arr)
   })
+
+  test('partition', () => {
+    expect(List.partition((x) => x > 1, [1, 2, 3])).toEqual(Tuple([2, 3], [1]))
+    expect(List.partition((x) => x > 1)([1, 2, 3])).toEqual(Tuple([2, 3], [1]))
+    expect(List.partition((x) => x > 1, [])).toEqual(Tuple([], []))
+    expect(List.partition((x) => x > 1)([])).toEqual(Tuple([], []))
+    // order preserved
+    expect(
+      List.partition((x) => x % 2 === 0, [1, 2, 3, 4, 5])
+    ).toEqual(Tuple([2, 4], [1, 3, 5]))
+  })
 })
diff --git a/src/List.ts b/src/List.ts
index 806888c..037b142 100644
--- a/src/List.ts
+++ b/src/List.ts
@@ -90,6 +90,36 @@ function sort<T>(compare: (a: T, b: T) => Order, list?: readonly T[]): any {
   }
 }
 
+/** Splits a list into two arrays based on a predicate. The first array contains the elements that satisfied the predicate and the second contains the rest */
+function partition<T>(
+  f: (x: T, index: number, arr: T[]) => boolean,
+  list: T[]
+): Tuple<T[], T[]>
+function partition<T>(
+  f: (x: T, index: number, arr: T[]) => boolean
+): (list: T[]) => Tuple<T[], T[]>
+function partition<T>(
+  f: (x: T, index: number, arr: T[]) => boolean,
+  list?: T[]
+): any {
+  switch (arguments.length) {
+    case 1:
+      return (list: T[]) => partition(f, list)
+    default: {
+      const accepted: T[] = []
+      const rejected: T[] = []
+      for (let i = 0; i < list!.length; i++) {
+        if (f(list![i], i, list!)) {
+          accepted.push(list![i])
+        } else {
+          rejected.push(list![i])
+        }
+      }
+      return Tuple(accepted, rejected)
+    }
+  }
+}
+
 export const List = {
   init,
   uncons,
@@ -100,5 +130,6 @@ export const List = {
   find,
   findIndex,
   sum,
-  sort
+  sort,
+  partition
 }
